db/schema_tables.cc

```diff
@@ -700,7 +700,9 @@ redact_columns_for_missing_features(mutation m, schema_features features) {
         return std::move(m);
     }
     slogger.debug("adjusting schema_tables mutation due to possible in-progress cluster upgrade");
-    m.upgrade(scylla_tables(features));
+    // The global schema ptr make sure it will be registered in the schema registry.
+    global_schema_ptr redacted_schema{scylla_tables(features)};
+    m.upgrade(redacted_schema);
     return std::move(m);
```

Comment:
Will it stay alive long enough?

Reply:
global_schema_ptr doesn't extend the life of a schema indefinitely.

Reply:
It will last until next reboot. The schema_ptr that it is registering is stored in a static thread local array in the scylla_tables function.

Reply:
Ah! yes.